src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1047,21 +1112,29 @@ QDateTime TorrentHandle::completedTime() const
         return QDateTime();
 }
 
-int TorrentHandle::timeSinceUpload() const
+std::chrono::seconds TorrentHandle::timeSinceUpload() const
 {
-    return m_nativeStatus.time_since_upload;
+#if LIBTORRENT_VERSION_NUM < 10200
+    return std::chrono::seconds(m_nativeStatus.time_since_upload);
+#else
+    return std::chrono::duration_cast<std::chrono::seconds>(libt::clock_type::now() - m_nativeStatus.last_upload);
```

Comment:
is this duration cast necessary? if you would return a `duration` I don't think it would be (as long as you end up using the same clock as libtorrent, which I think you'd be likely to)